ofrak_core/ofrak/model/component_filters.py

```diff
@@ -19,64 +20,66 @@ def _isinstance(*args, **kwargs):
     return isinstance(*args, **kwargs)
 
 
+@dataclass(frozen=True, init=False)
 class ComponentWhitelistFilter(ComponentFilter):
     """
     Only allow components which belong to a specific set to be run.
     """
 
-    whitelisted_component_ids: Set[bytes]
+    whitelisted_component_ids: FrozenSet[bytes]
 
-    def __init__(self, *whitelisted_component_ids: bytes):
-        self.whitelisted_component_ids = set(whitelisted_component_ids)
+    def __init__(self, *whitelisted_component_ids):
+        object.__setattr__(self, "whitelisted_component_ids", frozenset(whitelisted_component_ids))
```

Comment:
Now that thesse filters are dataclasses, are these `__init__`s still needed?

Reply:
To avoid changing all the usages of the constructors, yes. Otherwise they all needed to be updated to call `frozenset` or other changes... creates a much bigger and muddier changeset, as well as less readable usage code. I prefer to keep the init details contained here, and also leaves it more open to future implementation changes.

Reply:
Downside of this approach is that the constructor doesn't match typehints in dataclass definition. This might be something worth revisiting at some point.